doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,86 @@
+Dispatching
+===========
+
+The dispatching mechanism allows users to use alternative implementations of the algorithms
+available in scikit-image. The alternative implementations are provided by separate
+Python packages maintained outside of scikit-image. Example use-cases are support for array
+libraries other than Numpy or optimised implementations for particular hardware.
+
+This section of the documentation describes how to create an alternative implementation (a "backend")
+and how the dispatching mechanism in scikit-image works.
+
```

Comment:
```suggestion

.. important::

    Dispatching support and infrastructure is very experimental, in early
    development and is not ready for production. It's made available as an 
    early draft so that frontend and backend developers can collect experience
    with such a system. Expect the dispatching API to change a lot in the
    future.

```

Feel very welcome to tweak it further. :blush: